### ISSUE TITLE
Named exports when using CJS

### ISSUE THREAD

**Issue by laggingreflex** [purtuga/esm-webpack-plugin#11]
(this assumes #10 is merged)

If I have:

```js
exports.a = 1
```
I can't do:
```
import { a } from ...
``` 
it exports everything as a default export when source uses CJS. 

I have to do
```
import whole from ...
whole.a
```

**Comment by purtuga** [comment]
@laggingreflex I been looking into this and analyzing webpack's `module` class instance and I am not seeing a way to identify the exported members from a CJS module.  If you happen to figure that out, I would welcome a contribution.

I was playing with this a bit - trying to understand what happens when webpack processes your source in "mixed module mode" (`javascript/auto`) and the results seem to be that if the entry point is not an ES module that has explicit exports, then it will not work.  My suggestion would be for you to create an ESM module that imports all members from the CJS modules you want to export and then export them explicitly. 

Here is a quick example of what I tried:

file: `index.cjs`:
```javascript
exports.libA = require("./lib-a.cjs").libA;

exports.cjsIndex = function cjsIndex() {
    console.log("src-cjs/index.cjs loaded!");
}
```

file: `index-cjs-to-esm.js`:
```javascript
import {libA, cjsIndex } from "./index.cjs";

export {libA, cjsIndex};
```
> NOTE that I think this will only work if webpack is running with `javascript/auto` mode

And, when I ran a build, I got the ESM bundle with:

```javascript
//....
export const libA = LIB['libA'];
export const cjsIndex = LIB['cjsIndex'];**
```

Can you try this approach and see if works for you?

**Comment by purtuga** [comment]
FYI - pushed through [another change](https://github.com/purtuga/esm-webpack-plugin/commit/8c91dbc797a1d1269c415983fb561d7b81a1b1c5) (on top of yours) where the `library` var name is exposed as a named export (for the cases where the use of `default` exports might not be allowed or discouraged).

Also, added documentation around use of this lib with CJS: 
https://github.com/purtuga/esm-webpack-plugin#with-commonjs-project-individual-exports-are-not-available-in-the-output-esm-module

**Comment by laggingreflex** [comment]
Awesome. 

The approach you mentioned won't be much use to me in my particular use case though... I'll keep it mind though if it comes in handy. 

I did delve in as much as I could in Webpack to find a solution but I too couldn't find any. In fact I'm leaning towards the possibility that this may not be possible at all. Given that CJS allows setting `exports` anywhere instead of just at the top-level... it might not be feasible to extract these exports reliably at all, so it's no wonder that info doesn't exists.

I'm happy enough with plain CJS exports for now.

**Comment by purtuga** [comment]
cool - thanks for your help and contribution @laggingreflex .
I too don't think its possible for CJS - one can even override `exports` as in `module.exports = function(){}`.
I just published a new version (1.2.0) to NPM that includes the latest changes.
